tests/lib/rules/require-test-case-name.ts

```diff
@@ -0,0 +1,446 @@
+import { RuleTester } from 'eslint';
+
+import rule from '../../../lib/rules/require-test-case-name.ts';
+
+/**
+ * Returns the code for some valid test cases
+ * @param cases The code representation of valid test cases
+ * @returns Code representing the test cases
+ */
+function getTestCases(valid: string[], invalid: string[] = []): string {
+  return `
+    new RuleTester().run('foo', bar, {
+      valid: [
+        ${valid.join(',\n        ')},
+      ],
+      invalid: [
+        ${invalid.join(',\n        ')},
+      ]
+    });
+  `;
+}
+
+const alwaysError = {
+  messageId: 'nameRequiredAlways',
+};
+
+const objectsError = {
+  messageId: 'nameRequiredObjects',
+};
+
+const objectsWithConfigError = {
+  messageId: 'nameRequiredObjectsWithConfig',
+};
+
+const ruleTester = new RuleTester({
+  languageOptions: { sourceType: 'commonjs' },
+});
+ruleTester.run('test-case-name-property', rule, {
+  valid: [
```

Comment:
It would be even better if there were some tests covering non-string and non-object cases:

```
getTestCases(['foo']),
getTestCases(['{code: foo}']),
getTestCases(['foo()']),
getTestCases(['{code: foo()}']),
getTestCases(['`foo;${bar}`']),
```